components/datetime/src/format/time_zone.rs

```diff
@@ -37,11 +37,79 @@ where
     T: TimeZoneInput,
 {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write_pattern(self.time_zone_format, self.time_zone, f).map_err(|_| core::fmt::Error)
+        write_zone(self.time_zone_format, self.time_zone, f).map_err(|_| core::fmt::Error)
     }
 }
 
-// TODO(#622) Make this public once TimeZoneFormat is public.
+pub(crate) fn write_zone<T, W>(
```

Comment:
> suggestion

We used to have only `write_pattern()`, which was `pub(crate)` as the top-level abstraction under the public API.

But now we have:
```
Public API
└── write_zone()
    ├── write_pattern()
    └── write_config()
```

So we should be able to leave `write_zone()` as the only `pub(crate)` entrypoint, and make both `write_pattern()` and `write_config()` fully private (i.e. no public specifier at all: only `fn`).